src/stack.js

```diff
@@ -1,5 +1,43 @@
 'use strict'
 
+class Node {
+  constructor( data, link ) {
+    this.data = data
+    this.link = link
+  }
+}
+
 export default class Stack {
-  // your code here
+  constructor() {
+    this.head = null
+    this.size = 0
+  }
+
+  push( data ) {
+    this.head = new Node( data, this.head )
+    this.size++
+  }
+
+  pop() {
+    if ( this.head !== null ) {
+      let topElement = this.head.data
+      this.head = this.head.link
+      this.size--
+      return topElement
+    }
+    return null
+  }
+
+  peek() {
+    return this.head !== null ? this.head.data : null
```

Comment:
this could even just be
```javascript
return this.head
  ? this.head.data
  : null
```
since if the head is null, it will return false when invoked